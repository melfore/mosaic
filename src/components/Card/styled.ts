import {
  Box as MUIBox,
  Card as MUICard,
  CardActions as MUICardActions,
  CardContent as MUICardContent,
  CardHeader as MUICardHeader,
  styled,
} from "@material-ui/core";

export const ActionsWrapper = styled(MUIBox)(({ theme }) => ({
  "& > *:not(:first-child)": {
    marginLeft: `${theme.spacing(2)}px`,
  },
}));

export const StyledMUICard = styled(MUICard)(({ theme }) => ({
  margin: `${theme.spacing(2)}px`,
  width: "100%",
}));

export const StyledMUICardActions = styled(MUICardActions)(({ theme }) => ({
  alignItems: "center",
  display: "flex",
  justifyContent: "space-between",
  padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
}));

export const StyledMUICardContent = styled(MUICardContent)(({ theme }) => ({
  padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
}));

export const StyledMUICardHeader = styled(MUICardHeader)(({ theme }) => ({
  padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
  "& > .MuiCardHeader-content": {
    width: `calc(100% - ${theme.spacing(9)}px)`,
  },
  "& > .MuiCardHeader-content > span": {
    display: "inherit",
  },
}));
