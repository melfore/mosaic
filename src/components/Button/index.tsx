import React, { FC } from "react";
import MUIButton from "@material-ui/core/Button";
import Icon from "../Icon";
import { BaseIntlType } from "../../types/Base";
import { ButtonIconPosition, ButtonIconType, ButtonType, ButtonVariants } from "../../types/Button";
import withIntl from "../../utils/hocs/withIntl";

const getIcons = (dataCy: string, iconConfig?: ButtonIconType) => {
  const icons = { endIcon: undefined, startIcon: undefined };
  if (!iconConfig) {
    return;
  }

  const { name, position } = iconConfig;
  const icon = <Icon dataCy={`${dataCy}-icon`} name={name} />;
  if (position === ButtonIconPosition.right) {
    return {
      ...icons,
      endIcon: icon,
    };
  }

  return {
    ...icons,
    startIcon: icon,
  };
};

/**
 * Button component made on top of `@material-ui/core/Button`.
 */
const Button: FC<ButtonType> = ({
  dataCy = "button",
  elevated = false,
  icon = undefined,
  label = "",
  onClick,
  variant = ButtonVariants.contained,
  disabled = false,
}) => {
  return (
    <MUIButton
      color="primary"
      data-cy={dataCy}
      disableElevation={!elevated}
      onClick={onClick}
      variant={variant}
      {...getIcons(dataCy, icon)}
      disabled={disabled}
    >
      {label}
    </MUIButton>
  );
};

export const ButtonIntl: FC<ButtonType & BaseIntlType> = withIntl(Button);

export default Button;
