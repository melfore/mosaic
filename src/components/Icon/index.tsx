import React, { FC } from "react";
import { IconType, IconSize } from "../../types/Icon";
import { iconsCatalog } from "./utils";

const Icon: FC<IconType> = ({ dataCy, forwarded = {}, name, size = IconSize.default }) => {
  const icon = iconsCatalog[name];
  return React.cloneElement(icon, { ...forwarded, "data-cy": dataCy, fontSize: size });
};

export default Icon;
