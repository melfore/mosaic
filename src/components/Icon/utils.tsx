import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
