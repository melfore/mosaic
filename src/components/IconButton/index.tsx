import React, { FC } from "react";
import MUIIconButton from "@material-ui/core/IconButton";
import Icon from "../Icon";
import { IconButtonType } from "../../types/IconButton";
import { IconSize } from "../../types/Icon";
import { Color } from "../../types/Base";

/**
 * IconButton component made on top of `@material-ui/core/IconButton`.
 *
 * Add required documentation, use HTML tags if needed.
 */
const IconButton: FC<IconButtonType> = ({
  color = Color.primary,
  dataCy = "icon-button",
  icon,
  onClick,
  disabled = false,
  size = IconSize.default,
}) => {
  const _icon = <Icon dataCy={`${dataCy}`} name={icon} size={size} />;

  return (
    <MUIIconButton color={color} onClick={onClick} disabled={disabled}>
      {_icon}
    </MUIIconButton>
  );
};

export default IconButton;
