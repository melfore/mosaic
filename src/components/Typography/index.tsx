import React, { FC } from "react";
import MUITypography from "@material-ui/core/Typography";
import { BaseIntlType } from "../../types/Base";
import { TypographyType, TypographyVariants } from "../../types/Typography";
import withIntl from "../../utils/hocs/withIntl";
import { getBottomSpacing, getTruncate, VARIANT_COMPONENT_MAP } from "./utils";

/**
 * Typography component made on top of `@material-ui/core/Typography`.
 */
const Typography: FC<TypographyType> = ({
  bottomSpacing = undefined,
  dataCy = "typography",
  label,
  truncated = undefined,
  variant = TypographyVariants.body,
}) => {
  return (
    <MUITypography
      className={`data-cy-${dataCy}`}
      gutterBottom={getBottomSpacing(variant, bottomSpacing)}
      noWrap={getTruncate(variant, truncated)}
      variant={variant}
      variantMapping={VARIANT_COMPONENT_MAP}
    >
      {label}
    </MUITypography>
  );
};

export const TypographyIntl: FC<TypographyType & BaseIntlType> = withIntl(Typography);

export default Typography;
