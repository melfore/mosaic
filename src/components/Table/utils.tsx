import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
