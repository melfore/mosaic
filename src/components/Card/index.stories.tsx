import React from "react";
import { action } from "@storybook/addon-actions";
import { text, select } from "@storybook/addon-knobs";
import { Icons } from "../../types/Icon";
import { getDocsPageStructure, StoriesWrapper } from "../../utils/stories";
import Button from "../Button";
import IconButton from "../IconButton";
import Typography from "../Typography";
import Card from ".";

export default {
  title: "Card",
  component: Card,
  parameters: {
    ...getDocsPageStructure("Card", false),
  },
};

export const Canvas = () => (
  <Card
    actions={[
      <Button label="Share" onClick={action("Share Button")} />,
      <IconButton icon={Icons.add} onClick={action("Add action")} />,
    ]}
    collapsible={
      <Typography label="Traditional mosaics are made of small, flat, roughly square pieces of stone or glass of different colors, known as tesserae. Some floor mosaics are made of small rounded pieces of stone and called pebble mosaics. Mosaic skinning (covering objects with mosaic glass) is done with thin enameled glass and opaque stained glass. Modern mosaic art is made from any material in any size ranging from carved stone, bottle caps, and found objects." />
    }
    content={
      <Typography label="A mosaic is an artistic picture or design made out of any materials assembled together. Mosaic are used as decoration. Architects use mosaic murals for kitchen backsplash, shower wall and entry floor art. Mosaic Craft items are used as home decor. Cities often decorate public places such as parks with mosaic murals and sculptures." />
    }
    icon={select("icon", Icons, Icons.add)}
    subtitle={text("subTitle", "Mosaic Card Subtitle")}
    title={text("title", "Mosaic Card")}
  />
);

export const Basic = () => (
  <StoriesWrapper>
    <Card
      content={
        <Typography label="A mosaic is an artistic picture or design made out of any materials assembled together." />
      }
      title="Mosaic"
    />
  </StoriesWrapper>
);

export const WithAvatar = () => (
  <StoriesWrapper>
    <Card
      content={<Typography label="Opening hours: 9AM - 6PM" />}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    />
  </StoriesWrapper>
);

export const WithActions = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      content={<Typography label="Opening hours: 9AM - 6PM" />}
      icon={Icons.business}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    />
  </StoriesWrapper>
);

export const WithCollapsible = () => (
  <StoriesWrapper>
    <Card
      actions={[<Button icon={{ name: Icons.open_new }} label="Discover" onClick={action("Discover")} />]}
      collapsible={<Typography label="Discover our catalog!" />}
      icon={Icons.business}
      content={<Typography label="Opening hours: 9AM - 6PM" />}
      subtitle="The best in town for components"
      title="Mosaic Shop"
    />
  </StoriesWrapper>
);
