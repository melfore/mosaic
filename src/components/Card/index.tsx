import React, { cloneElement, FC, Fragment, useState, useMemo } from "react";
import { Collapse as MUICollapse, useTheme } from "@material-ui/core";
import { Skeleton as MUISkeleton } from "@material-ui/lab";
import { CardType } from "../../types/Card";
import { Icons } from "../../types/Icon";
import { TypographyVariants } from "../../types/Typography";
import Avatar from "../Avatar";
import IconButton from "../IconButton";
import Typography from "../Typography";
import {
  ActionsWrapper,
  StyledMUICard,
  StyledMUICardActions,
  StyledMUICardContent,
  StyledMUICardHeader,
} from "./styled";

/**
 * Card component made on top of `@material-ui/core/Card`
 */
const Card: FC<CardType> = ({
  actions = [],
  children,
  collapsible = undefined,
  icon = undefined,
  loading = false,
  title,
  subtitle = undefined,
  unmountCollapsible = false,
}) => {
  const [expanded, setExpanded] = useState(false);
  const theme = useTheme();

  const cardHeader = useMemo(
    () => (
      <StyledMUICardHeader
        avatar={icon && <Avatar icon={icon} loading={loading} />}
        disableTypography
        title={
          <Typography
            bottomSpacing={false}
            label={title}
            loading={loading}
            truncated
            variant={TypographyVariants.title}
          />
        }
        subheader={
          <Typography
            bottomSpacing={false}
            label={subtitle}
            loading={loading}
            truncated
            variant={TypographyVariants.caption}
          />
        }
      />
    ),
    [icon, loading, subtitle, title]
  );

  return (
    <StyledMUICard>
      {cardHeader}
      <StyledMUICardContent>
        {loading ? <MUISkeleton height={`${theme.spacing(16)}px`} /> : children}
      </StyledMUICardContent>
      {!loading && (
        <Fragment>
          <StyledMUICardActions disableSpacing>
            {actions.length > 0 && (
              <ActionsWrapper alignItems="center" display="flex">
                {actions.map((action, index) => cloneElement(action, { key: `card-action-${index}` }))}
              </ActionsWrapper>
            )}
            {collapsible && (
              <IconButton icon={expanded ? Icons.up : Icons.down} onClick={() => setExpanded(!expanded)} />
            )}
          </StyledMUICardActions>
          {collapsible && (
            <MUICollapse in={expanded} timeout="auto" unmountOnExit={unmountCollapsible}>
              <StyledMUICardContent>{collapsible}</StyledMUICardContent>
            </MUICollapse>
          )}
        </Fragment>
      )}
    </StyledMUICard>
  );
};

export default Card;
