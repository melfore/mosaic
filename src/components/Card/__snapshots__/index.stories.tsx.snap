// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surfaces/Card Actions smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <div style="align-items: center; display: flex;">
      <div style="margin-right: 16px;">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
                tabindex="0"
                type="button"
                data-cy="button"
        >
          <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="OpenInNewIcon"
                 data-cy="button-icon"
            >
              <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
              </path>
            </svg>
          </span>
          Gallery
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card Collapsible play-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
            tabindex="0"
            type="button"
            data-cy="card-collapse"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="KeyboardArrowUpIcon"
           data-cy="card-collapse-icon"
      >
        <path d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z">
        </path>
      </svg>
    </button>
  </div>
  <div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-mui-classname"
       style="min-height: 0px; height: auto; transition-duration: 196ms;"
  >
    <div class="MuiCollapse-wrapper MuiCollapse-vertical css-mui-classname">
      <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mui-classname">
        <div class="MuiCardContent-root css-mui-classname"
             data-cy="card-collapsible-content"
             style="padding: 8px 16px;"
        >
          Discover mosaics from all around the globe!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card CollapsibleActions smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
    <div style="align-items: center; display: flex;">
      <div style="margin-right: 16px;">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
                tabindex="0"
                type="button"
                data-cy="button"
        >
          <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
            <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
                 focusable="false"
                 aria-hidden="true"
                 viewbox="0 0 24 24"
                 data-testid="OpenInNewIcon"
                 data-cy="button-icon"
            >
              <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
              </path>
            </svg>
          </span>
          Gallery
        </button>
      </div>
    </div>
    <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
            tabindex="0"
            type="button"
            data-cy="card-collapse"
    >
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="KeyboardArrowDownIcon"
           data-cy="card-collapse-icon"
      >
        <path d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z">
        </path>
      </svg>
    </button>
  </div>
  <div class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-mui-classname"
       style="min-height: 0px;"
  >
    <div class="MuiCollapse-wrapper MuiCollapse-vertical css-mui-classname">
      <div class="MuiCollapse-wrapperInner MuiCollapse-vertical css-mui-classname">
        <div class="MuiCardContent-root css-mui-classname"
             data-cy="card-collapsible-content"
             style="padding: 8px 16px;"
        >
          Discover mosaics from all around the globe!
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Surfaces/Card Icon smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-avatar css-mui-classname">
      <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-mui-classname"
           data-cy="card-avatar"
           style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="AppsIcon"
             data-cy="card-avatar-icon"
        >
          <path d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z">
          </path>
        </svg>
      </div>
    </div>
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;

exports[`Surfaces/Card Loading smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-avatar css-mui-classname">
      <span class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto css-mui-classname">
        <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-mui-classname"
             data-cy="card-avatar-loading"
        >
          <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-mui-classname"
               focusable="false"
               aria-hidden="true"
               viewbox="0 0 24 24"
               data-testid="PersonIcon"
          >
            <path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
            </path>
          </svg>
        </div>
      </span>
    </div>
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title-loading"
      >
        <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-mui-classname">
        </span>
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle-loading"
      >
        <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-mui-classname">
        </span>
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    <span class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-mui-classname"
          style="height: 128px;"
    >
    </span>
  </div>
</div>
`;

exports[`Surfaces/Card Localized smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="locale.title"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="locale.title-title"
      >
        Titolo
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="locale.title-subtitle"
      >
        Sottotitolo
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="locale.title-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;

exports[`Surfaces/Card Primary smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mui-classname"
     data-cy="card"
     style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); margin: 16px; width: calc(100% - 32px);"
>
  <div class="MuiCardHeader-root css-mui-classname">
    <div class="MuiCardHeader-content css-mui-classname">
      <h2 class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-mui-classname"
          data-cy="card-title"
      >
        Mosaic
      </h2>
      <span class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-mui-classname"
            data-cy="card-subtitle"
      >
        UK: /mə(ʊ)ˈzeɪɪk/
      </span>
    </div>
  </div>
  <div class="MuiCardContent-root css-mui-classname"
       data-cy="card-content"
       style="padding: 8px 16px;"
  >
    A mosaic is an artistic picture or design made out of any materials assembled together.
  </div>
  <div class="MuiCardActions-root css-mui-classname"
       style="align-items: center; display: flex; justify-content: space-between; padding: 8px 16px;"
  >
  </div>
</div>
`;
